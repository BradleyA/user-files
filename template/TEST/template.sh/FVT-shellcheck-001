#!/bin/bash
#	template/TEST/template.sh/FVT-shellcheck-001  3.260.500  2019-08-14T00:15:54.361251-05:00 (CDT)  https://github.com/BradleyA/user-files.git  uadmin  one-rpi3b.cptx86.com 3.259  
#	   testing shellcheck 
#	template/TEST/template.sh/FVT-shellcheck-001  3.251.485  2019-08-11T22:55:55.438285-05:00 (CDT)  https://github.com/BradleyA/user-files.git  uadmin  one-rpi3b.cptx86.com 3.250-4-geba20b1  
#	   template/TEST/template.sh/FVT-shellcheck-001 added shellscript test case 
### production standard 3.0 shellcheck
### production standard 5.1.160 Copyright
#       Copyright (c) 2019 Bradley Allen
#       MIT License is in the online DOCUMENTATION, DOCUMENTATION URL defined below.
BOLD=$(tput -Txterm bold)
NORMAL=$(tput -Txterm sgr0)
###	Place test case here

#	analyse code for potential errors
shellcheck "${1}" > "${0}".test-case-output

###
diff "${0}".expected "${0}".test-case-output >/dev/null 2>&1
RETURN_CODE=${?}
if [ ${RETURN_CODE} -eq 0 ] ; then
	echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - No difference with expected output - ${BOLD}PASS - PASS${NORMAL}"
elif [ ${RETURN_CODE} -eq 1 ] ; then
	echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - Differences with expected output - ${BOLD}FAIL - FAIL${NORMAL}"
	diff -y "${0}".expected "${0}".test-case-output
	exit 1
else 
	echo "${BOLD}Test case --->${NORMAL} ${0} ${1} ${RETURN_CODE} - Test script ERROR - ${BOLD}FAIL - FAIL${NORMAL}"
	exit 2
fi
