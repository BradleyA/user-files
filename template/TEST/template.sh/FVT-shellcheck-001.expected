
In ../../template.sh line 27:
DEFAULT_SSH_USER="${USER}"
^-- SC2034: DEFAULT_SSH_USER appears unused. Verify it or export it.


In ../../template.sh line 28:
DEFAULT_TLS_USER="${USER}"
^-- SC2034: DEFAULT_TLS_USER appears unused. Verify it or export it.


In ../../template.sh line 29:
DEFAULT_ADM_TLS_USER="${USER}"
^-- SC2034: DEFAULT_ADM_TLS_USER appears unused. Verify it or export it.


In ../../template.sh line 34:
DEFAULT_CERT_DIR="${HOME}/.docker"
^-- SC2034: DEFAULT_CERT_DIR appears unused. Verify it or export it.


In ../../template.sh line 35:
DEFAULT_CERT_DAEMON_DIR="/etc/docker/certs.d/daemon/"
^-- SC2034: DEFAULT_CERT_DAEMON_DIR appears unused. Verify it or export it.


In ../../template.sh line 36:
DEFAULT_CA_CERT="ca.pem"
^-- SC2034: DEFAULT_CA_CERT appears unused. Verify it or export it.


In ../../template.sh line 37:
DEFAULT_CA_PRIVATE_CERT="ca-priv-key.pem"
^-- SC2034: DEFAULT_CA_PRIVATE_CERT appears unused. Verify it or export it.


In ../../template.sh line 38:
DEFAULT_NUMBER_DAYS="730"
^-- SC2034: DEFAULT_NUMBER_DAYS appears unused. Verify it or export it.


In ../../template.sh line 39:
DEFAULT_REMOTE_COMMAND=""
^-- SC2034: DEFAULT_REMOTE_COMMAND appears unused. Verify it or export it.


In ../../template.sh line 40:
DEFAULT_REMOTE_COMMAND_OPTION=""
^-- SC2034: DEFAULT_REMOTE_COMMAND_OPTION appears unused. Verify it or export it.


In ../../template.sh line 41:
DEFAULT_WORKING_DIRECTORY="$(echo ~/.docker/docker-ca)"
^-- SC2034: DEFAULT_WORKING_DIRECTORY appears unused. Verify it or export it.
                           ^-- SC2116: Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.


In ../../template.sh line 42:
DEFAULT_FQDN="$(hostname -f)"		# local host
^-- SC2034: DEFAULT_FQDN appears unused. Verify it or export it.


In ../../template.sh line 43:
DEFAULT_REMOTE_HOST="$(hostname -f)"    # local host
^-- SC2034: DEFAULT_REMOTE_HOST appears unused. Verify it or export it.


In ../../template.sh line 44:
DEFAULT_REGISTRY_HOST="$(hostname -f)"    # local host
^-- SC2034: DEFAULT_REGISTRY_HOST appears unused. Verify it or export it.


In ../../template.sh line 45:
DEFAULT_REGISTRY_PORT="5000"
^-- SC2034: DEFAULT_REGISTRY_PORT appears unused. Verify it or export it.


In ../../template.sh line 289:
SCRIPT_NAME=$(head -2 "${0}" | awk {'printf $2'})
                                   ^-- SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
                                               ^-- SC1083: This } is literal. Check expression (missing ;/\n?) or quote it.


In ../../template.sh line 290:
SCRIPT_VERSION=$(head -2 "${0}" | awk {'printf $3'})
                                      ^-- SC1083: This { is literal. Check expression (missing ;/\n?) or quote it.
                                                  ^-- SC1083: This } is literal. Check expression (missing ;/\n?) or quote it.


In ../../template.sh line 323:
if ! [ $(id -u) = 0 ] ; then
       ^-- SC2046: Quote this to prevent word splitting.


In ../../template.sh line 413:
                        SSH_USER=$(echo "${1}" | cut -d '=' -f 2)
                        ^-- SC2034: SSH_USER appears unused. Verify it or export it.


In ../../template.sh line 426:
                        USER_HOME=$(echo "${1}" | cut -d '=' -f 2)
                        ^-- SC2034: USER_HOME appears unused. Verify it or export it.

